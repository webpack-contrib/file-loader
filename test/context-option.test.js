import path from 'path';

import {
  compile,
  execute,
  getCompiler,
  normalizeErrors,
  readAsset,
} from './helpers';

describe('"context" option', () => {
  it('should work without value', async () => {
    const compiler = getCompiler('simple.js', {
      name: '[path][name].[ext]',
      context: __dirname,
    });
    const stats = await compile(compiler);

    expect(
      execute(readAsset('main.bundle.js', compiler, stats))
    ).toMatchSnapshot('result');
    expect(normalizeErrors(stats.compilation.warnings)).toMatchSnapshot(
      'warnings'
    );
    expect(normalizeErrors(stats.compilation.errors)).toMatchSnapshot('errors');
  });

  it('should work with "String" value', async () => {
    const compiler = getCompiler('simple.js', {
      name: '[path][name].[ext]',
      context: path.resolve(__dirname, '../'),
    });
    const stats = await compile(compiler);

    expect(
      execute(readAsset('main.bundle.js', compiler, stats))
    ).toMatchSnapshot('result');
    expect(normalizeErrors(stats.compilation.warnings)).toMatchSnapshot(
      'warnings'
    );
    expect(normalizeErrors(stats.compilation.errors)).toMatchSnapshot('errors');
  });
});
