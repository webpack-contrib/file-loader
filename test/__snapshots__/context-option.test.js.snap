// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`"context" option should work with "String" value: errors 1`] = `Array []`;

exports[`"context" option should work with "String" value: result 1`] = `"test/fixtures/file.png"`;

exports[`"context" option should work with "String" value: warnings 1`] = `Array []`;

exports[`"context" option should work without value: errors 1`] = `Array []`;

exports[`"context" option should work without value: result 1`] = `"fixtures/file.png"`;

exports[`"context" option should work without value: warnings 1`] = `Array []`;
